### ISSUE TITLE
issue with textareaGrow() when changing textarea content via javascript

### ISSUE THREAD

**Issue by lsmith77** [maoberlehner/replicating-the-twitter-tweet-box-with-vue#1]
first up thank you for this amazing tutorial!

I have integrated part of your code into my Vue app but I run into issues with the `scrollHeight` not updating in case I change the contents of the textarea programmatically.

Do you maybe have an idea on how to cover this case?

**Comment by lsmith77** [comment]
actually let me build a reproducible test case in a PR

**Comment by lsmith77** [comment]
alright .. so the issue is that I am relying on `watch` which you recommend against, since I am using a two-way binding into the vuex store to update the textarea.

a quick ugly ugly hack fix

```
        watch: {
            textValue() {
                setTimeout(this.textareaGrow, this.$nextTick);
            },
        },
```

but I guess ideally I need to somehow either move `textareaGrow()` logic to the store or well not sure.

**Comment by maoberlehner** [comment]
Yeah, you have to use `watch`. I also did solve something similar using `watch` in one of my apps.
